src/engine/sync/enginesync.cpp

```diff
@@ -32,17 +38,76 @@ EngineSync::EngineSync(UserSettingsPointer pConfig)
 EngineSync::~EngineSync() {
 }
 
+Syncable* EngineSync::pickMaster(Syncable* enabling_syncable) {
+    // If there is an explicit master, and it is playing, keep it.
+    if (m_pMasterSyncable && m_pMasterSyncable->getSyncMode() == SYNC_MASTER_EXPLICIT && m_pMasterSyncable->isPlaying()) {
+        return m_pMasterSyncable;
+    }
+
+    static std::vector<Syncable*> stopped_sync_decks;
+    static std::vector<Syncable*> playing_sync_decks;
+    stopped_sync_decks.reserve(kMaxNumberOfDecks);
```

Comment:
reserve() allocates

Reply:
please propose a solution for out-of-thread allocation. I am tired of arguing over 16 bytes

Reply:
I do not plan to argue. Dynamically allocating memory in real-time threads is a no go, even a single byte. Why not use static allocation if you know the limits upfront?

Reply:
do you mean a plain array?  I could then DEBUG_ASSERT on n > kMaxNumberOfDecks but it makes me sad to add a runtime fail condition because we are trying to avoid dynamic containers

Reply:
I think we can get rid of all of them. It should be possible to break the loops early and return a result without store it as first()

Reply:
oh!  I think you're right. Trying that now